tests/load_lpal_again.py

```diff
@@ -0,0 +1,126 @@
+
+# happens twice in main(run()), don't need double the amount of code
+# returns 0 for no errors and 1 for errors
+def integrityCheck(straw, connection, numErrors, isInitital):
+    initOrNo = "Initial" if isInitital else "Secondary"
+    # check integrity of this straw in straw_present
+    integrity = s_util.checkStrawIntegrity(straw,connection)
+    # checkStrawIntegrity returns 0 if all is good, 1-3 for error codes
+    if integrity==0:
+        logger.info(f"{initOrNo} integrity check: {straw.id} passes.")
+    elif integrity == 1:
+        logger.error(f"{initOrNo} integrity check: {straw.id} has multiple of the same locaiton.")
+        numErrors += 1
+        return 1
+    elif integrity == 2:
+        logger.error(f"{initOrNo} integrity check: {straw.id} is present in multiple locations.")
+        numErrors += 1
+        return 1
+    elif integrity == 3:
+        logger.error(f"{initOrNo} integrity check: {straw.id} has conflicting timestamps.")
+        numErrors += 1
+        return 1
+    # no problems found
+    return 0
+
+
+
+def run(lpal_file):
+
+    # make sure user wants to proceed
+    logger.info('Warning: This script will potentially modify the database.  Enter "Y" to continue.')
+    response = input()
+    if not(response == "Y" or response =="y"):
+        logger.info("Script aborted.")
+        return
+
+
+    # get rows and fields from lpal file
+    rows, fields = read_file(lpal_file)
+
+    # connect to DB
+    engine, connection = connectDB()
+
+    # get LPAL location ID
+    lpalLoc = StrawLocation.query().filter(StrawLocation.location_type == 'LPAL').filter(StrawLocation.number == int((sys.argv[1])[4:8]))
+
+    # collect number of errors
+    numErrors = 0
+
+    for row in rows:
+        #logger.info(f'Straw{(row["Straw"])[2:]}')
+        # get the number and make sure it exists
+        straw = s_util.get_or_create_straw((row["Straw"])[2:])
+
+        # check integrity on straw_present
+        if integrityCheck(straw, connection, numErrors, True):
+            # error found, skip straw
+            continue
+        
+        # search for any location that is a panel
+        inPanel = any(str(e).find("MN") for e in straw.locations())
+        inLPAL = any(str(e).find("LPAL") for e in straw.locations())
+        
+
+        # if in a panel get the in time
+        if inPanel:
+            panelEntry = straw.currentPresence()
+            # if straw isn't present here
+            if panelEntry[2] is not 1:
+                # throw error or something
+                continue
+            # otherwise get the in time to be submitted later
+            panelInTime = panelEntry[3]
+        
+        if not inLPAL:
+            logger.info(f"Submitting straw ST{straw.id} to DB with values:")
+            logger.info(f"Straw: {straw.id}, Position: {lpalLoc}, Present: {not inPanel}, time_in: {row['Timestamp']}, time_out: {panelInTime if inPanel else None}")
+            result=s_util.newEntry(
```

Comment:
`StrawPresent(a,b,c,d,...).commit()`

Reply:
will need the `StrawPosition`, `straw_position = StrawPosition(straw_location_object, position_number)` or maybe `GetStrawPosition(straw_location_number, straw_location_pallet_id, position_number)`.